Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm

```diff
@@ -288,20 +289,47 @@
     m_replacedDocumentNodes.remove(uuid);
     m_replacementLocationOffsets.remove(uuid);
 
-    removeTransparentMarkersForSession(uuid, *sessionRange);
+    removeTransparentMarkersForSession(uuid, true);
 }
 
-void UnifiedTextReplacementController::removeTransparentMarkersForSession(const WTF::UUID& uuid, WebCore::SimpleRange& range)
+void UnifiedTextReplacementController::removeTransparentMarkersForSession(const WTF::UUID& uuid, bool removeAll)
 {
     RefPtr document = this->document();
     if (!document) {
         ASSERT_NOT_REACHED();
         return;
     }
 
-    document->markers().filterMarkers(range, [&] (const WebCore::DocumentMarker& marker) {
-        return std::get<WebCore::DocumentMarker::TransparentContentData>(marker.data()).uuid == uuid ? WebCore::FilterMarkerResult::Remove : WebCore::FilterMarkerResult::Keep;
-    }, { WebCore::DocumentMarker::Type::TransparentContent });
+    if (auto sessionRange = contextRangeForSessionWithUUID(uuid)) {
+        document->markers().filterMarkers(*sessionRange, [&](const WebCore::DocumentMarker& marker) {
```

Comment:
This code seems to be incredibly repetitive, can you factor it out into a lambda or something?